superset/utils/database.py

```diff
@@ -39,10 +39,7 @@ def get_or_create_db(
     from superset.models import core as models
 
     database = (
-        db.session.query(models.Database)
-        .filter_by(database_name=database_name)
-        .autoflush(False)
-        .first()
+        db.session.query(models.Database).filter_by(database_name=database_name).first()
```

Comment:
Although the Select can also be placed in a transaction, it relies on the subsequent commit(), which is not useful.